plugins/reolink/src/main.ts

```diff
@@ -295,6 +316,21 @@ class ReolinkCamera extends RtspSmartCamera implements Camera, DeviceProvider, R
             && this.storageSettings.values.abilities?.value?.Ability?.supportAudioAlarm?.ver !== 0;
     }
 
+    hasFloodlight() {
+        const channel = this.getRtspChannel();
+
+        const channelData = this.storageSettings.values.abilities?.value?.Ability?.abilityChn?.[channel];
+        if (channelData) {
+            const floodLightConfigVer = channelData.floodLight?.ver ?? 0;
```

Comment:
you need optional chaining on channelData.

Reply:
It's checked in the if, isn't it?